modules/tests/js/src/test/scala/coursier/Platform.scala

```diff
@@ -15,12 +15,21 @@ object Platform {
 
   val timeout = 4000
 
+  // https://www.scala-js.org/doc/interoperability/global-scope.html#dynamically-lookup-a-global-variable-given-its-name
+  private def globalObject =
+    if (js.typeOf(g.global) != "undefined" && (g.global.Object eq g.Object))
+      // Node.js environment detected
+      g.global
+    else
+      // In all other well-known environment, we can use the global `this`
+      js.special.fileLevelThis.asInstanceOf[js.Dynamic]
+
   /** Available if we're running on node, and package xhr2 is installed */
   lazy val xhr =
-    if (js.isUndefined(g.XMLHttpRequest))
-      g.require("xhr2")
+    if (js.isUndefined(globalObject.XMLHttpRequest))
+      globalObject.require("xhr2")
     else
-      g.XMLHttpRequest
+      globalObject.XMLHttpRequest
```

Comment:
Of course, the same story applies here, and everywhere else where this pattern comes up.